### PR TITLE
SD-470: handle errors and edge cases for faucet

### PR DESCRIPTION
**Pull request by domagojlatecki**

### CODE REVIEW

File: src/test/kotlin/com/ampnet/identityservice/service/FaucetQueueServiceTest.kt [hunk 1]
```diff
@@ -75,7 +110,89 @@ class FaucetQueueServiceTest : TestBase() {
...
+        }
+
+        verify("Task will fail and new task will be created") {
+            processAllTasks(maxAttempts = 1)
+
+            val tasks = faucetTaskRepository.findAll()
+
+            assertThat(tasks).hasSize(2)
+            assertThat(tasks.first().status).isEqualTo(FaucetTaskStatus.FAILED)
+            assertThat(tasks[1].status).isEqualTo(FaucetTaskStatus.CREATED)
```
Comment (githinho) [inline src/test/kotlin/com/ampnet/identityservice/service/FaucetQueueServiceTest.kt:140]: Maybe remove `first` because it is tricky to read with number `1` below:
```suggestion
            assertThat(tasks.[0].status).isEqualTo(FaucetTaskStatus.FAILED)
            assertThat(tasks[1].status).isEqualTo(FaucetTaskStatus.CREATED)
```

File: src/test/kotlin/com/ampnet/identityservice/service/FaucetQueueServiceTest.kt [hunk 2]
```diff
@@ -75,7 +110,89 @@ class FaucetQueueServiceTest : TestBase() {
...
+
+        verify("Task will fail and new task will be created") {
+            processAllTasks(maxAttempts = 1)
+
+            val tasks = faucetTaskRepository.findAll()
+
+            assertThat(tasks).hasSize(2)
+            assertThat(tasks.first().status).isEqualTo(FaucetTaskStatus.FAILED)
+            assertThat(tasks.first().hash).isEqualTo(hash)
+            assertThat(tasks[1].status).isEqualTo(FaucetTaskStatus.CREATED)
```
Comment (githinho) [inline src/test/kotlin/com/ampnet/identityservice/service/FaucetQueueServiceTest.kt:162]: Also, first and 1.

File: src/test/kotlin/com/ampnet/identityservice/service/FaucetQueueServiceTest.kt [hunk 3]
```diff
@@ -159,14 +324,15 @@ class FaucetQueueServiceTest : TestBase() {
...
+        verify("Task will fail and new task will be created") {
+            processAllTasks(maxAttempts = 1)
 
             val tasks = faucetTaskRepository.findAll()
 
-            assertThat(tasks).hasSize(1)
+            assertThat(tasks).hasSize(2)
             assertThat(tasks.first().status).isEqualTo(FaucetTaskStatus.FAILED)
             assertThat(tasks.first().hash).isEqualTo(hash)
+            assertThat(tasks[1].status).isEqualTo(FaucetTaskStatus.CREATED)
```
Comment (githinho) [inline src/test/kotlin/com/ampnet/identityservice/service/FaucetQueueServiceTest.kt:335]: First and 1.

File: src/test/kotlin/com/ampnet/identityservice/service/BlockchainQueueServiceTest.kt [hunk 4]
```diff
@@ -89,43 +97,134 @@ class BlockchainQueueServiceTest : TestBase() {
...
+
+        verify("Task will fail and new task will be created") {
+            processAllTasks(maxAttempts = 1)
+
+            val tasks = blockchainTaskRepository.findAll()
+
+            assertThat(tasks).hasSize(1)
+            assertThat(tasks.first().status).isEqualTo(BlockchainTaskStatus.FAILED)
+            assertThat(tasks.first().hash).isEqualTo(hash)
+        }
```
Comment (githinho) [inline src/test/kotlin/com/ampnet/identityservice/service/BlockchainQueueServiceTest.kt:158]: Looks like verify description is incorrect because there is only one task in repo.

File: src/main/kotlin/com/ampnet/identityservice/config/ApplicationProperties.kt [hunk 5]
```diff
@@ -67,3 +70,14 @@ class PinataProperties {
     var jwt: String = ""
     var maxUses: Int = 10
 }
+
+@Suppress("MagicNumber")
+class FaucetProperties {
+    var maxAddressesPerTask: Int = 100
+    var gasLimit: BigInteger = BigInteger.valueOf(4_000_000) // test run with 100 random addresses used 3,829,595 gas
```
[the remark this answers is not in the document]
Comment (domagojlatecki) [reply src/main/kotlin/com/ampnet/identityservice/config/ApplicationProperties.kt:77]: Yes, it works as intended.

**Review by githinho — APPROVED** [review]
